modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/artifact/AbstractParameterResolverExecutor.java

```diff
@@ -73,7 +80,8 @@ protected ParameterValueResolver parameterValueResolver(ParameterizedElementDecl
   }
 
   protected Map<String, Object> parametersMap(ParameterizedElementDeclaration parameterizedElementDeclaration,
-                                              ParameterizedModel parameterizedModel)
+                                              ParameterizedModel parameterizedModel,
+                                              boolean overrideExpressionSupport)
```

Comment:
I would say that if we go with expressions for ValueProvider body level, we would need it for SampleData too and later for Metadata resolution. Therefore, I would say this by default the new behaviour.

Reply:
Maybe we can change that in the future. I would not want to accept expressions for ValueProviders yet

Reply:
Ok, makes sense.